runtime/runtime.go

```diff
@@ -1475,15 +1475,37 @@ func (r *interpreterRuntime) standardLibraryFunctions(
 	interpreterOptions []interpreter.Option,
 	checkerOptions []sema.Option,
 ) stdlib.StandardLibraryFunctions {
+	builtins := stdlib.FlowBuiltInFunctions(stdlib.FlowBuiltinImpls{
+		CreateAccount:   r.newCreateAccountFunction(context, storage, interpreterOptions, checkerOptions),
+		GetAccount:      r.newGetAccountFunction(context.Interface, storage),
+		Log:             r.newLogFunction(context.Interface),
+		GetCurrentBlock: r.newGetCurrentBlockFunction(context.Interface),
+		GetBlock:        r.newGetBlockFunction(context.Interface),
+		UnsafeRandom:    r.newUnsafeRandomFunction(context.Interface),
+	})
+
+	switch context.Location.(type) {
+	case common.ScriptLocation:
+		// Scripts are read-only, so we can give them access to auth accounts
+		builtins = append(builtins,
+			stdlib.NewStandardLibraryFunction(
+				"getAuthAccount",
+				&sema.FunctionType{
+					Parameters: []*sema.Parameter{{
+						Label:          sema.ArgumentLabelNotRequired,
+						Identifier:     "address",
+						TypeAnnotation: sema.NewTypeAnnotation(&sema.AddressType{}),
+					}},
+					ReturnTypeAnnotation: sema.NewTypeAnnotation(sema.AuthAccountType),
+				},
```

Comment:
Maybe factor this our into a global, so we don't have to allocate it again on each call